migrations/versions/0323_broadcast_message.py

```diff
@@ -0,0 +1,94 @@
+"""
+
+Revision ID: 0323_broadcast_message
+Revises: 0322_broadcast_service_perm
+Create Date: 2020-07-02 11:59:38.734650
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.sql import column, func
+from sqlalchemy.dialects import postgresql
+
+from app.models import BroadcastMessage
+
+revision = '0323_broadcast_message'
+down_revision = '0322_broadcast_service_perm'
+
+
+name = 'template_type'
+tmp_name = 'tmp_' + name
+
+old_options = ('sms', 'email', 'letter')
+new_options = old_options + ('broadcast',)
+
+new_type = sa.Enum(*new_options, name=name)
+old_type = sa.Enum(*old_options, name=name)
+
+
+STATES = [
+    'draft',
+    'pending-approval',
+    'rejected',
+    'broadcasting',
+    'completed',
+    'cancelled',
+    'technical-failure',
+]
+states_with_quotes_and_commas = ', '.join(f"'{x}'" for x in STATES)
+
+
+def upgrade():
+    op.execute(f'ALTER TYPE {name} RENAME TO {tmp_name}')
```

Comment:
Historically we the db user for our applications didn't have permission to `alter type {name} add value {value}`
I wonder if that is still true.

Reply:
just tried it out on preview (creating a new enum and table) and it works fine.

last time we tried one of these was april 2017 (https://github.com/alphagov/notifications-api/blob/master/migrations/versions/0078_add_sent_notification_status.py). I can't get the exact date but it looks like we'd been live on paas for a month or so at that point (https://github.com/alphagov/notifications-api/pull/870)

Reply:
update: misread this, learnt about `add value`, updated the upgrade script

Reply:
update update: You can't run `alter table {name} add value {value}` from within a transaction. (This is changed in postgres v12)
This means the migration script isn't repeatable if it fails. I don't really like that, I would rather keep the existing way.